src/geom/intersection_tools.C

```diff
@@ -76,6 +78,71 @@ bool collinear(const Point & p1,
   return true;
 }
 
+bool within_edge_on_side(const Elem & elem,
+                         const Point & p,
+                         const unsigned short s,
+                         ElemExtrema & extrema,
+                         const bool linearize,
+                         const Real tol)
+{
+  libmesh_assert_less(s, elem.n_sides());
+  libmesh_assert(extrema.is_invalid());
+  libmesh_assert_equal_to(elem.dim(), 3);
+
+  // For higher order than linear without linearization, make sure
+  // that our edges are collinear
+  if (elem.default_order() > 1 && !linearize)
+    for (const auto e : elem.edge_index_range())
+    {
+      // we should expect 3 edges for our higher order elems
```

Comment:
3-noded edges

Reply:
And actually, could you just assert that `n_nodes_on_edge > 3`?  We do hope to support higher-order Lagrange and Rational Bezier-Bernstein one of these days.